packages/base-driver/lib/jsonwp-proxy/proxy.js

```diff
@@ -130,7 +147,12 @@ class JWProxy {
       // we have something like /session/:id/foobar, so we need to replace
       // the session id
       const match = sessionBaseRe.exec(remainingUrl);
-      remainingUrl = remainingUrl.replace(match[1], this.sessionId);
+      // TODO: if `requiresSessionId` is `false` and `sessionId` is `null`, this is a bug.
+      // are we sure `sessionId` is not `null`?
```

Comment:
i think the `.test()` just above guarantees this?

Reply:
It's a bug because you can't (read: _shouldn't_, especially as per TypeScript) attempt to replace a substring with `null`. e.g., `'dull'.replace('dull', null)`

Reply:
yeah i was thinking of something else but you are right, maybe there's not a guarantee that `this.sessionId` is set here. at any rate it doesn't appear to be a common bug or we'd have run into it. maybe we should add a check and throw an exception so if it does happen in the future we at least know what's up?

Reply:
My best guess is that we haven't hit the bug because given the happy path, `sessionBaseRe.test(remainingUrl)` is never `true` if there's no session.  That may just mean someone hasn't tried to do something weird with an endpoint--like specify a session ID in the URL when `this.sessionId` is `null`.

So I think then it makes sense to throw if `sessionBaseRe.test(remainingUrl)` is `true` and `sessionId` is `null`.